### ISSUE TITLE
Indexed access on Java objects

### ISSUE THREAD

**Issue by anri-vin** [caer/v8-adapter#30]
Hello.

 I want to use indexed/key access on Java objects which are not maps, lists or arrays. In Rhino it could be done by overriding `public Object get(int index); public Object get(String key);` on wrapper objects. Is there any way to achieve same with your library? Sorry if it mentioned somewhere, but I could not find an answer. 

Thank you.

**Comment by crahda** [comment]
Hi @anri-vin ,

Thanks for reaching out! When you say indexed/key access, do you mean that, given an instance of the following Java class:

```java
public class MyJavaClass {
    public String myString = "Hello, World!";
    public int myInteger = 1337;
}
```

The following JavaScript code:

```js
console.log(myJavaClassInstance["myString"]);
console.log(myJavaClassInstance.myInteger);
```

Would output the following to the console:

```shell
Hello, World!
1337
```

If so, I do not *think* we have a built-in feature for this; @AlexTrotsenko may be able to chime in, as he has added quite a few features along these lines in the past. If we do not have a feature for this, there are probably a few ways to implement one, but I will not be able to take charge of that at the moment.

**Comment by AlexTrotsenko** [comment]
@crahda yes it's implemented. But for encapsulation reasons it uses get/set property _(as @anri-vin  actually posted)_

Check https://github.com/AlexTrotsenko/v8-adapter/blob/870e306dbb0b5d6194a2edceff64ee4d25bad4c5/src/test/java/io/alicorn/v8/V8JavaAdapterTest.java#L615

It can be replaced with 

```
"var x = new WannabeBeanNoAnnotations(); x['i'] = 6688; x.i;"
```

**Comment by anri-vin** [comment]
I think we are not completely understanding each other. What I want is when I have Java class like this:

```
class JavaObject {
   private Object[] internalArray = ...;
    
   public Object getProperty(int index) {
      return internalArray[index]; 
   }
   public void setProperty(int index, Object value) {
      internalArray[index] = value;
   }
}
```
calling JS as following
`javaObject[42]; javaObject[42] = 0;`
will internally call getProperty and setProperty.

**Comment by AlexTrotsenko** [comment]
@anri-vin as far as I know - this behaviour with indexed setter is not implemented. Since I have not used this one in Rhino either - it's a bit hard to tell you want should be the easiest way to achieve this _(as I am not sure about exact use case)_.

I would start looking in the `io.alicorn.v8.V8JavaClassInterceptor` api. If it's not matching your needs - then as @crahda mentioned - you can probably try implement one yourself. Normally @crahda is quite fast to reviewing PR.

Also if you are using latest version of J2V8 and thus newer V8 itself - you can likely use [JS Proxy](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Proxy) api in you app and redirect requests to the _set/get_ methods of your object _(actually Js proxy object of your Java object)_. I was experimenting with it in [this PR](https://github.com/alicorn-systems/v8-adapter/pull/12/files#diff-179e305970c610f5e50d1275175a4223R39), but since proxies are not supported in J2V8 `4.6.0` _(current target of this lib)_ - I skipped this change for now.

**Comment by caer** [comment]
FYI: With the upstream project being largely unmaintained for ~4 years, and my own lack of maintenance on the project, I'm closing all open tickets and archiving this repository. 💔
